### PR TITLE
Define changes for transitions, two bug fixes

### PR DESCRIPTION
**Pull request by Lilaa3**
Fixed higher time transitions not working (up to 255 frames right now) and fixed transitions not starting properly when interrupting another transition, cleaned up a few things, put every non vanilla styled change into POLISHED_TRANSITIONS and renamed set_and_reset_transition_fade_timer
Arthurtilly and Artic wanted this reverted by default, i think a "correct" full radius counts as a fix and smoothstep fixes the "ended before you'd expect" look that linear has, but I understand now why maybe it should be left commented out by default.
I tried also adjusting the times with POLISHED_TRANSITIONS but it looked very out of place, more than some faster looking transitions did.

### CODE REVIEW

File: src/game/screen_transition.c [hunk 1]
```diff
@@ -24,7 +24,7 @@ void *sTextureTransitionID[] = {
 };
 
 
-s32 set_and_reset_transition_fade_timer(u8 transTime) {
+s32 set_and_reset_transition_timers(u8 transTime) {
```
Comment (gheskett) [inline src/game/screen_transition.c:27]: This is only resetting one timer.

**Review by gheskett — APPROVED** [review]
Moving this to 2.1.3

File: src/game/screen_transition.c (hunk 1 above)
Comment (Lilaa3) [reply src/game/screen_transition.c:27]: It also resets the angle, how about set_and_reset_transitions_values?

Comment (gheskett) [reply src/game/screen_transition.c:27]: I think old name was probably fine tbh

**Pull request merged**